File: ChainOfResponsibility/C++/DebugLogger.cpp
#include "DebugLogger.hpp"
#include <iostream>

DebugLogger::DebugLogger(Logger* nextLoggerProcessor) : Logger(nextLoggerProcessor)
{
}

void DebugLogger::logMessage(int logLevel, std::string message)
{
    if (logLevel == Logger::DEBUG)
    {
        std::cout << "DEBUG: " << message << std::endl;
    }
    else
    {
        Logger::logMessage(logLevel, message);
    }
}

File: ChainOfResponsibility/Java/Logger/DebugLogger.java
public class DebugLogger extends Logger {
    DebugLogger(Logger nextLogProcessor) {
        super(nextLogProcessor);
    }

    @Override
    public void log(int logLevel, String message) {
        if (logLevel == DEBUG) {
            System.out.println("DEBUG: " + message);
        }
        if (nextLogProcessor != null) {
            nextLogProcessor.log(logLevel, message);
        }
    }
}